temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -193,6 +194,13 @@ private RetryDecision shouldRetry(
       return new RetryDecision(RetryState.RETRY_STATE_NON_RETRYABLE_FAILURE, null);
     }
 
+    if (executionFailure != null
+        && executionFailure.hasApplicationFailureInfo()
+        && RetryOptionsUtils.isNotRetryable(
+            retryOptions, executionFailure.getApplicationFailureInfo().getType())) {
+      return new RetryDecision(RetryState.RETRY_STATE_NON_RETRYABLE_FAILURE, null);
+    }
```

Comment:
Consider assigning this and the expressions above which all result in the same `RETRY_STATE_NON_RETRYABLE_FAILURE` decision  to bools and just or-ing them all together so there's only one return path per unique decision.

Reply:
Not possible, this specific order is important to make sense of what's going on. The RETRY_STATE_RETRY_POLICY_NOT_SET outcome has to divide two code paths leading to RETRY_STATE_NON_RETRYABLE_FAILURE for different reasons.